src/core/pointcloud/qgslazdecoder.cpp

```diff
@@ -315,7 +315,9 @@ void decodePoint( char *buf, int lasPointFormat, char *dataBuffer, std::size_t &
   lazperf::las::nir14 nir;
   lazperf::las::point14 p14;
 
-  bool isLas14 = ( lasPointFormat == 6 || lasPointFormat == 7 || lasPointFormat == 8 );
+  // Does the point record start with the common fields for formats introduced
+  // in the LAS 1.4 spec?
+  bool isLas14 = ( lasPointFormat == 6 || lasPointFormat == 7 || lasPointFormat == 8 || lasPointFormat == 9 || lasPointFormat == 10 );
```

Comment:
pdrf 9 and 10 are not supported so this is kind of redundant. You could also `const` it while there :)

Reply:
side remark from someone ignorant about the LAS spec. Using named constant to explain the semantics of 6, 7, etc. would be better for readability (assuming there's some semantic behind those numbers...)

Reply:
> pdrf 9 and 10 are not supported so this is kind of redundant. You could also `const` it while there :)

Const added :)

I kept formats 9 and 10, since I think the variable is less confusing this way, actually checking for format commonalities and not for (common format && supported format).



> side remark from someone ignorant about the LAS spec. Using named constant to explain the semantics of 6, 7, etc. would be better for readability (assuming there's some semantic behind those numbers...)

The spec just numbers the possible formats, so I don't think adding enums would gain us much here. Besides, this is just a bugfix PR, and refactoring this everywhere would increase its scope too much.